src/contracts/core/AllocationManager.sol

```diff
@@ -490,15 +508,18 @@ contract AllocationManager is
         StrategyInfo memory info,
         Allocation memory allocation
     ) internal {
-        // Update encumbered magnitude
-        encumberedMagnitude[operator][strategy] = info.encumberedMagnitude;
-        emit EncumberedMagnitudeUpdated(operator, strategy, info.encumberedMagnitude);
+        // Update encumbered magnitude if it has changed
+        // The mapping should NOT be updated when there is a deallocation on a delay
+        if (encumberedMagnitude[operator][strategy] != info.encumberedMagnitude) {
+            encumberedMagnitude[operator][strategy] = info.encumberedMagnitude;
+            emit EncumberedMagnitudeUpdated(operator, strategy, info.encumberedMagnitude);
+        }
 
         // Update allocation for this operator set from the strategy
+        // We emit an `AllocationUpdated` from the `modifyAllocations` and `slashOperator` functions.
+        // `clearDeallocationQueue` does not emit an `AllocationUpdated` event since it was
+        // emitted when the deallocation was queued
```

Comment:
Ah this comment makes it clear to me why you removed the event from this internal function